cpp/src/parquet/encryption/encryption.h

```diff
@@ -46,28 +48,56 @@ using ColumnPathToEncryptionPropertiesMap =
 
 class PARQUET_EXPORT DecryptionKeyRetriever {
  public:
-  virtual std::string GetKey(const std::string& key_metadata) = 0;
+  /// \brief Retrieve a key.
+  /// \deprecated Deprecated since 21.0.0.
```

Comment:
Ahem. 21.0.0 is nearly released, so we should actually rename all these deprecation notices to say 22.0.0.